etna/pipeline/base.py

```diff
@@ -591,20 +591,23 @@ def _validate_backtest_stride(n_folds: Union[int, List[FoldMask]], horizon: int,
         return stride
 
     @staticmethod
-    def _validate_backtest_dataset(
-        ts: TSDataset, n_folds: int, horizon: int, stride: int
-    ):  # TODO: try to optimize, works really slow on datasets with large number of segments
+    def _validate_backtest_dataset(ts: TSDataset, n_folds: int, horizon: int, stride: int):
         """Check all segments have enough timestamps to validate forecaster with given number of splits."""
         min_required_length = horizon + (n_folds - 1) * stride
-        segments = set(ts.df.columns.get_level_values("segment"))
-        for segment in segments:
-            segment_target = ts[:, segment, "target"]
-            if len(segment_target) < min_required_length:
-                raise ValueError(
-                    f"All the series from feature dataframe should contain at least "
-                    f"{horizon} + {n_folds-1} * {stride} = {min_required_length} timestamps; "
-                    f"series {segment} does not."
-                )
+
+        df = ts.df.loc[:, (slice(None), "target")]
+        num_timestamps = df.shape[0]
+        not_na = ~np.isnan(df.values)
+        min_idx = np.argmax(not_na, axis=0)
```

Comment:
Are you sure it works fine if there are no non-nan values? I think in that case the check should fail.

Reply:
I guess it works fine, it just return 0 in case of no non-nan
<img width="542" alt="Снимок экрана 2023-12-14 в 13 06 26" src="https://github.com/etna-team/etna/assets/55380696/f362dd6b-ed45-4df9-ad55-69ed8ae4b3cd">

Reply:
We don't actually have tests for this method

Reply:
I think that length of empty segment should be treated as 0. Probably, we should fail in this case. What do you think?